README.md
<img src="http://davinci.sazonov.org/images/davinci-logo.svg" width="300">

**Davinci** is a scientific software for the visualization and processing of single-crystal diffraction data measured with a point detector.

Please, visit [davinci.sazonov.org](http://davinci.sazonov.org) for more details.

<!---https://github.com/dwyl/repo-badges--->

[![][16]][11] [![][17]][12] [![][18]][13] [![][28]][22]

[![][31]][30] [![][45]][41] [![][52]][50]

<!---Appveyor CI Build Status--->
[10]: https://ci.appveyor.com/project/AndrewSazonov/davinci
[11]: https://ci.appveyor.com/project/AndrewSazonov/davinci?branch=windows_mingw
[12]: https://ci.appveyor.com/project/AndrewSazonov/davinci?branch=windows_msvc
[13]: https://ci.appveyor.com/project/AndrewSazonov/davinci?branch=linux_gcc
[15]: https://ci.appveyor.com/api/projects/status/605mg4m6l3qfoe6f/branch/master?svg=true
[16]: https://img.shields.io/appveyor/ci/AndrewSazonov/Davinci/windows_mingw.svg?label=Windows%20MinGW
[17]: https://img.shields.io/appveyor/ci/AndrewSazonov/Davinci/windows_msvc.svg?label=Windows%20MSVC
[18]: https://img.shields.io/appveyor/ci/AndrewSazonov/Davinci/linux_gcc.svg?label=Linux%20GCC

<!---Travis CI Build Status--->
[20]: https://travis-ci.org/AndrewSazonov/Davinci
[21]: https://travis-ci.org/AndrewSazonov/Davinci/branches
[22]: https://travis-ci.org/AndrewSazonov/Davinci/branches
[25]: https://travis-ci.org/AndrewSazonov/Davinci.svg?branch=master
[26]: https://img.shields.io/travis/AndrewSazonov/Davinci.svg?label=Linux%20%26%20macOS
[27]: https://img.shields.io/travis/AndrewSazonov/Davinci/linux_gcc.svg?label=Linux%20GCC
[28]: https://img.shields.io/travis/AndrewSazonov/Davinci/macos_clang.svg?label=macOS%20Clang

<!---GitHub Latest Release--->
[30]: https://github.com/AndrewSazonov/Davinci/releases/latest
[31]: https://img.shields.io/github/release/AndrewSazonov/Davinci.svg?label=Release
[32]: https://img.shields.io/badge/Release-v1.0.0-blue.svg

<!---User Manual--->
[40]: https://andrewsazonov.gitbooks.io/davinci/content/
[41]: http://davinci.sazonov.org/umanual.html
[45]: https://img.shields.io/badge/User_Manual-30.05.2018-blue.svg
[46]: Resources/Shields/User_Manual.svg

<!---License--->
[50]: https://github.com/AndrewSazonov/Davinci/blob/master/LICENSE
[51]: https://img.shields.io/github/license/AndrewSazonov/Davinci.svg?label=License
[52]: https://img.shields.io/badge/License-GNU_GPL_v3.0-blue.svg

<!---Coverity--->
[60]: https://scan.coverity.com/projects/andrewsazonov-davinci
[61]: https://scan.coverity.com/projects/9665/badge.svg


<!---
!
https://www.gitbook.com/book/0xax/linux-insides/details
!
http://showdownjs.com/
https://github.com/saucelabs-ansible/pip/edit/master/README.md
https://github.com/standard/standard

https://guides.github.com/features/mastering-markdown/
https://github.com/TimothyYe/skm/edit/master/README.md
https://github.com/buildkite/dynamic-build-badges
https://github.com/kevinoid/travis-status
https://github.com/TimothyYe/skm/edit/master/README.md
https://github.com/didi/cube-ui
https://github.com/laravel/laravel
https://github.com/apache/storm
https://github.com/JedWatson/sydjs-site
https://github.com/spring-io/sagan

Build all branches in travis or set up a cron (available in travis) to build the master branch daily.

--->


CHANGELOG.md
# Davinci Version History

## Version 1.0.7 (19 Aug 2018)
* Adds progress bar to give an indication of how long the data processing is going to take
* Adds more multi-threading features in order to spead up the data processing
* Fixes an issue with ShelX output format in the console version of the program
* Fixes multiple other issues

## Version 1.0.6 (09 Jul 2018)
* Adds reading of Psi angle from the NICOS instrument data collection files (Issue #5)
* Adds UMWEG output for the calculation of multiple-diffraction patterns
* Adds some multi-threading features in order to spead up the data treatment

## Version 1.0.5 (30 May 2018)
* Adds calculation of the direction cosines (Issue #3)
* Adds reading of Psi angle from the HEiDi instrument data collection files
* Adds natural sort order and sort by date and time to the table widgets
* Adds ShelX output with real (non-integer) Miller indices hkl
* Fixes calculation of the Phi angle in the 4-circle geometry
* Fixes an issue with Tbar/D9 output (theta + temperature, psi, fwhm)
* Fixes some minor issues

## Version 1.0.4 (28 Mar 2018)
* Adds support of the new input format for POLI: polarized NICOS data with adet
* Adds user manual and report issue links to the Help menu
* Adds a program icon and desktop file for the Linux version

## Version 1.0.3 (13 Mar 2018)
* Fixes issue #2, Not reading new files with Oxford magnet on POLI

## Version 1.0.2 (27 Jan 2018)
* Fixes a bug with the update function of the installer parogram

## Version 1.0.1 (11 Jan 2018)
* Adds support of the new input format: HEiDi at MLZ instrument log data
* Fixes an issue with automatic determination of the peak and background parameters
* Fixes a bug with cursor position change in the input data viewer when go through the scans
* Fixes a crash when open a new file
* Fixes a sidebar issue with the tabs width

## Version 1.0.0 (09 Dec 2017)
* First release



Scripts/changelog.py
#!/usr/bin/env python3

from functions import * 

# ChangelogArticle
class ChangelogArticle:

    __slots__ = ['version', 'date', 'list']

    def __init__(self, version='', date='', list=[]):
        self.version = version
        self.date = date
        self.list = list

# Changelog
class Changelog:
    def __init__(self, title=''):
        #self.app = 'Davinci'
        self.title = title
        self.last_article = ChangelogArticle()
        self.all_articles = []
        self.idx = 0
        self.add_articles()

    def __iter__(self):
        return self

    def __next__(self):
        self.idx += 1
        try:
            return self.all_articles[self.idx-1]
        except IndexError:
            self.idx = 0
            raise StopIteration

    def __getitem__(self, idx):
        return self.all_articles[idx]

    def version(self):
        return self.last_article.version

    def date(self):
        return self.last_article.date
        #return ConvertDate(self.last_article.date, '%Y-%m-%d', '%d %b %Y')

    def add(self, version, date, list):
        date = ConvertDate(date, '%Y-%m-%d', '%d %b %Y')
        self.last_article = ChangelogArticle(version, date, list)
        self.all_articles.insert(0, self.last_article)
        #self.all_articles.append(self.last_article)

    def add_articles(self):

        self.add(version="1.0.0",
                 date="2017-12-09",
                 list=[ 'First release' ])

        self.add(version="1.0.1",
                 date="2018-01-11",
                 list=[ 'Adds support of the new input format: HEiDi at MLZ instrument log data',
                        'Fixes an issue with automatic determination of the peak and background parameters',
                        'Fixes a bug with cursor position change in the input data viewer when go through the scans',
                        'Fixes a crash when open a new file',
                        'Fixes a sidebar issue with the tabs width' ])

        self.add(version="1.0.2",
                 date="2018-01-27",
                 list=[ 'Fixes a bug with the update function of the installer parogram' ])

        self.add(version="1.0.3",
                 date="2018-03-13",
                 list=[ 'Fixes issue #2, Not reading new files with Oxford magnet on POLI' ])

        self.add(version="1.0.4",
                 date="2018-03-28",
                 list=[ 'Adds support of the new input format for POLI: polarized NICOS data with adet',
                        'Adds user manual and report issue links to the Help menu',
                        'Adds a program icon and desktop file for the Linux version' ])

        self.add(version="1.0.5",
                 date="2018-05-30",
                  list=[ 'Adds calculation of the direction cosines (Issue #3)', # update manual
                         'Adds reading of Psi angle from the HEiDi instrument data collection files', # update manual
                         'Adds natural sort order and sort by date and time to the table widgets', # update manual
                         'Adds ShelX output with real (non-integer) Miller indices hkl',
                         'Fixes calculation of the Phi angle in the 4-circle geometry',
                         'Fixes an issue with Tbar/D9 output (theta + temperature, psi, fwhm)',
                         'Fixes some minor issues' ])

        self.add(version="1.0.6",
                 date="2018-07-09",
                  list=[ 'Adds reading of Psi angle from the NICOS instrument data collection files (Issue #5)', # update manual
                         'Adds UMWEG output for the calculation of multiple-diffraction patterns',
                         'Adds some multi-threading features in order to spead up the data treatment' ]) # update manual

        self.add(version="1.0.7",
                 date="2018-08-19",
                  list=[ 'Adds progress bar to give an indication of how long the data processing is going to take',
                         'Adds more multi-threading features in order to spead up the data processing',
                         'Fixes an issue with ShelX output format in the console version of the program',
                         'Fixes multiple other issues' ]) # update manual
